### ISSUE TITLE
Gitter Web Interface: accessibility improvements

### ISSUE THREAD

**Issue by zersiax** [gitterHQ/gitter#538]
Gitter.im , as of now, is reasonably usable by people who use a screenreader. A number of small issues persist however, that when fixed, would make the experience at least five times as good. Let me list some stumbling blocks I have ran into and my proposed fixes. Full disclosure, I am a blind computer science student myself.

* As of now, the settings icon is completely invisible to me. I am guessing it is some sort of icon or image that lacks an alt description and is not generated in the usual way, since I don't see an unlabeled graphic either which could be that particular icon. See if this control can be labeled properly by either adding alt text, a label or a role attribute.
* Reading through chat messages can sometimes suddenly jump you around rather eratically. I am guessing this is something to do with gitter intercepting arrow keys to do its own thing, are arrow keys mapped to actions in Gitter?
* As of now, notifications are invisible to screenreaders unless you happen to be at the (for a screenreader) bottom of the page, since it seems a DOM node is appended to the very end of the HTML hierarchy to denote a new chat has been received. Turning this into a live region would help immensely in detecting incoming private messages or chats in other channels.
* As of now, the semantic structure of pages is a little lacking. A heading is often used to denote a new, important section on a website. HTML5 sections and articles, as well as div roles have given web developers an alternative to this, but perhaps it would be good to add some more semantics to gitter.im pages to differentiate between the list of channels, chat messages area and text entry field.
* Currently, the only way to see what channel you are in as a screenreader user seems to be checking the title bar of the browser, or reading through chat history and inferring it from that . I am not sure how this is visually indicated, if at all.

These are the things I've ran into. As I said, gitter.im can currently be used by the blind, but it's a bit of a chore sometimes and applying these fixes would help immensely.

**Comment by malditogeek** [comment]
Hey @zersiax, thanks a lot for reporting this! I will take a look at this soon. Can you please tell me what screen reader are you using? I'll set it up to see how we can improve it. I'm on Mac so I'll set up VoiceOver to start with. Cheers!

**Comment by zersiax** [comment]
Hi,
I
am currently using NVDA on Windows with the latest Firefox. If you
want to test, that screenreader is open-source so you can just
download it and take it for a spin :)



2015-01-23 21:52 GMT+01:00, Mauro Pompilio <notifications@github.com>:

**Comment by agauniyal** [comment]
@zersiax  These are brilliant advices even for non-gitter related stuff. Never paid attention to them , will now.

**Comment by MarcoZehe** [comment]
Hi! I am also a blind screen reader user interested in using Gitter more. I was wondering what the status of this issue is, since there hasn't been any activity on it in a long time. Also, some of the related issues like the one about native links seem like low-hanging fruit whose fixes will already add hugely to the usability of Gitter. Please let us know if you need any help with these issues!

**Comment by MadLittleMods** [comment]
@MarcoZehe We are currently working on a11y improvements and roadmap to tackle these issues.

**Comment by MarcoZehe** [comment]
Well, the issues @zersiax mentions in his original report above still apply, and the problems with the iOS app that @stevefaulkner filed in another bug, are also still present. These are pretty good guidelines already. ;) Also, I just joined your support channel (through the IRC bridge, so can be pinged.
